### PR TITLE
Allow log_domain to be null.

### PR DESCRIPTION
**Pull request by chengchangwu**
Hope this has fixed issue #28.

### CODE REVIEW

File: glib/src/log.rs [hunk 1]
```diff
@@ -114,27 +114,39 @@ fn to_log_flags(fatal: bool, recursion: bool) -> u32 {
...
         log_domain: *const libc::c_char,
         log_level: glib_sys::GLogLevelFlags,
         message: *const libc::c_char,
         user_data: glib_sys::gpointer,
     ) {
-        let log_domain: Borrowed<GString> = from_glib_borrow(log_domain);
         let message: Borrowed<GString> = from_glib_borrow(message);
         let callback: &P = &*(user_data as *mut _);
-        (*callback)(log_domain.as_str(), from_glib(log_level), message.as_str());
+        if log_domain.is_null() {
```
Comment (sdroege) [inline glib/src/log.rs:135]: You can do `let log_domain: Borrowed<Option<GString>> = from_glib_borrow(log_domain)` and then use it as `log_domain.as_deref()` or similar

Comment (sdroege) [reply glib/src/log.rs:135]: No need for the separate `NULL` check

**Review by sdroege — COMMENTED** [review]
Looks good to me otherwise, thanks!

File: glib/src/log.rs (hunk 1 above)
Comment (chengchangwu) [reply glib/src/log.rs]: I've fixed that in the next commit.

**Comment by chengchangwu** [comment]
Why can the github pipeline report more errors than I can find with `cargo test`.

**Comment by chengchangwu** [comment]
I do not know how to change type of `log_domain` in `g_log` macro and fix the errors reported by `cargo test`.

File: glib/src/log.rs [hunk 2]
```diff
@@ -304,30 +318,30 @@ pub fn log_default_handler(log_domain: &str, log_level: LogLevel, message: Optio
 /// ```no_run
 /// use glib::{LogLevel, g_log};
 ///
-/// g_log!("test", LogLevel::Debug, "test");
-/// g_log!("test", LogLevel::Message, "test");
+/// g_log!(Some("test"), LogLevel::Debug, "test");
```
Comment (sdroege) [inline glib/src/log.rs:321]: For the macro there should simply be another variant that works without domain so you can do

```rust
g_log!(LogLevel::Debug, "test");
g_log!("domain", LogLevel::Debug, "test");
```

Comment (sdroege) [reply glib/src/log.rs:321]: @GuillaumeGomez Can you help with that maybe?

Comment (GuillaumeGomez) [reply glib/src/log.rs:321]: Taking a look!

**Comment by GuillaumeGomez** [comment]
Updated!

File: glib/src/log.rs [hunk 3]
```diff
@@ -331,17 +331,68 @@ pub fn log_default_handler(log_domain: Option<&str>, log_level: LogLevel, messag
...
 /// g_log!(Some("test"), LogLevel::Warning, "test: {} {}", x, "a",);
 /// ```
+///
+/// To be noted that the log domain is optional:
+///
+/// ```no_run
+/// use glib::{LogLevel, g_log};
+///
+/// // As you can see: no log domain:
+/// g_log!("test", LogLevel::Message, "test");
```
Comment (sdroege) [inline glib/src/log.rs:341]: You actually provide a log domain here :)

File: glib/src/log.rs [hunk 4]
```diff
@@ -581,10 +632,10 @@ macro_rules! g_info {
 /// ```
 #[macro_export]
 macro_rules! g_debug {
-    ($log_domain:expr, $format:expr, $($arg:expr),* $(,)?) => {{
+    ($log_domain:expr, $format:literal, $($arg:expr),* $(,)?) => {{
```
Comment (sdroege) [inline glib/src/log.rs:635]: The examples in the docs above everywhere still use `Some("test")` as log domain. That should allow omitting the log domain completely, or passing it as `"test"` without the `Option` wrapping :)

File: glib/src/log.rs (hunk 3 above)
Comment (GuillaumeGomez) [reply glib/src/log.rs:341]: Woups! Good catch. :)

File: glib/src/log.rs (hunk 4 above)
Comment (GuillaumeGomez) [reply glib/src/log.rs:635]: Oh right, it's clearly not mandatory.

Comment (GuillaumeGomez) [reply glib/src/log.rs:635]: Well, not anymore at least.

**Comment by GuillaumeGomez** [comment]
I removed `Some` from the examples.

**Pull request merged**